src/helpers/user_normalizer.coffee

```diff
@@ -0,0 +1,7 @@
+class UserNormalizer
+  @normalize: (username, userInput)->
+    if not userInput? or (userInput?.toLocaleLowerCase() is 'me')
+      return '@' + username
+    '@' + userInput.replace /@*/g, ''
```

Comment:
Fair enough, however Slack would not be the only use case for this. I think a better option would be to have a config environment variable (e.g. `HUBOT_TEAM_NORMALIZE_USERNAMES` or something like that) that enables the normalizing of user names -- default being false.